mm2src/mm2_main/src/rpc.rs

```diff
@@ -269,26 +271,48 @@ async fn rpc_service(req: Request<Body>, ctx_h: u32, client: SocketAddr) -> Resp
     // https://github.com/artemii235/SuperNET/issues/219
     let rpc_cors = match ctx.conf["rpccors"].as_str() {
         Some(s) => try_sf!(HeaderValue::from_str(s)),
-        None => HeaderValue::from_static("http://localhost:3000"),
+        None => {
+            if ctx.is_https() {
+                HeaderValue::from_static("https://localhost:3000")
+            } else {
+                HeaderValue::from_static("http://localhost:3000")
+            }
+        },
     };
 
     // Convert the native Hyper stream into a portable stream of `Bytes`.
     let (req, req_body) = req.into_parts();
+
+    if req.method == Method::OPTIONS {
+        return Response::builder()
+            .status(StatusCode::OK)
+            .header(ACCESS_CONTROL_ALLOW_ORIGIN, rpc_cors)
+            .header("Access-Control-Allow-Methods", "POST, OPTIONS")
```

Comment:
nit: we don't need to specify that we allow `OPTIONS` in the response to an `OPTIONS` request :)

Reply:
I used the example here https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request and `OPTIONS` was part of `Access-Control-Allow-Methods` for the example Preflight request.